mobile/src/main/java/org/openhab/habdroid/core/UpdateBroadcastReceiver.kt

```diff
@@ -170,26 +170,6 @@ class UpdateBroadcastReceiver : BroadcastReceiver() {
                     }
                 }
             }
-            if (comparableVersion <= WIDGETS_NO_AUTO_GEN_LABEL) {
-                val widgetComponent = ComponentName(context, ItemUpdateWidget::class.java)
-                AppWidgetManager.getInstance(context).getAppWidgetIds(widgetComponent).forEach { id ->
-                    val oldData = ItemUpdateWidget.getInfoForWidget(context, id)
-
-                    val newData = ItemUpdateWidget.ItemUpdateWidgetData(
-                        oldData.item,
-                        oldData.command,
-                        oldData.label,
-                        oldData.widgetLabel
-                            ?: context.getString(R.string.item_update_widget_text, oldData.label, oldData.mappedState),
-                        oldData.mappedState,
-                        oldData.icon,
-                        oldData.theme,
-                        oldData.showState
-                    )
-
-                    ItemUpdateWidget.saveInfoForWidget(context, newData, id)
-                }
-            }
```

Comment:
What's the point of removing that migration code now?

Reply:
`oldData.widgetLabel` cannot be null anymore, so `?: context.getString(R.string.item_update_widget_text, oldData.label, oldData.mappedState)` has to be removed.
Then the migration would be a no-op.

Reply:
Isn't the migration code just buggy then? Reading the existing data must happen taking the old format into account (which by definition can not change), not the new one.

Reply:
> Reading the existing data must happen taking the old format into account

Sure I can do that. But I wondering how long `widgetLabel` should be kept nullable only because of the old format.

Reply:
> But I wondering how long widgetLabel should be kept nullable only because of the old format.

It doesn't need to be. It's just the conversion code that needs to be able to consume a null value. Storing the data and the app widget code itself doesn't need to deal with null values.